packages/grid/x-data-grid/src/components/cell/GridEditInputCell.tsx

```diff
@@ -92,15 +92,31 @@ const GridEditInputCell = React.forwardRef<HTMLInputElement, GridEditInputCellPr
           await onValueChange(event, newValue);
         }
 
-        setValueState(newValue);
-        apiRef.current.setEditCellValue({ id, field, value: newValue, debounceMs }, event);
+        const column = apiRef.current.getColumn(field);
+
+        let parsedValue = newValue;
+        if (column.valueParser && rootProps.experimentalFeatures?.newEditingApi) {
+          parsedValue = column.valueParser(newValue, apiRef.current.getCellParams(id, field));
```

Comment:
Since we don't sync the value between the global state and the local state, when the call is debounced, we need to manually call the value parser and skip the one in `apiRef.current.setEditCellValue`.